### PR TITLE
Enable trivy scan in CI

### PR DESCRIPTION
**Pull request by liuw1**

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
+        with:
+          scan-type: 'fs'
+          scanners: 'vuln'
+          ignore-unfixed: true
```
Comment (xiaoyuxlu) [inline .github/workflows/trivy.yml:39]: why ignore unfixed?

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    name: Build
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
```
Comment (xiaoyuxlu) [inline .github/workflows/trivy.yml:35]: why use this commit( 2 years ago)? why not use the latest one (0.24.0 2weeks ago)?

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (liuw1) [reply .github/workflows/trivy.yml]: Removed

File: .github/workflows/trivy.yml (hunk 2 above)
Comment (liuw1) [reply .github/workflows/trivy.yml]: Good catch, updated.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**